modules/opensearch/opensearch.client.inc

```diff
@@ -355,28 +355,48 @@ function opensearch_execute(TingClientRequest $request) {
     $request->setRelationData('full');
   }
 
-  // Build cache id for this request.
-  $cid = _opensearch_cache_id($request);
-
   // Check static cache first as this is the fasts lookup. We use the fast
   // static pattern described on drupal.org in the drupal_static
   // documentation.
   static $opensearch_static_fast = array();
-  if (array_key_exists($cid, $opensearch_static_fast)) {
-    return $opensearch_static_fast[$cid];
-  }
 
-  // Check the database/memcached cache for the request also check that the
-  // cache data has not expired before using it.
-  $cache = cache_get($cid, 'cache_opensearch');
-  if ($cache && ($cache->expire > REQUEST_TIME)) {
-    $data = $cache->data;
-
-    // Update the static cache with the database cache content to speed up the
-    // next request for the same data.
-    $opensearch_static_fast[$cid] = $data;
+  if ($response = _opensearch_cache_get($request, $opensearch_static_fast)) {
```

Comment:
Maybe a comment that if an exact match for the $request is found fast-track and return the result.